src/tiled/editableobject.cpp

```diff
@@ -80,4 +85,90 @@ bool EditableObject::checkReadOnly() const
     return false;
 }
 
+static Map *mapForObject(Object *object)
+{
+    if (!object)
+        return nullptr;
+
+    switch (object->typeId()) {
+    case Object::LayerType:
+        return static_cast<Layer*>(object)->map();
+    case Object::MapObjectType:
+        return mapForObject(static_cast<MapObject*>(object)->objectGroup());
```

Comment:
This feels like a property MapObject should have.

Reply:
Yeah, I guess there are lots of places where we want to get the map of an object already.